Sources/Sentry/SentryTimeToDisplayTracker.m

```diff
@@ -0,0 +1,109 @@
+#import "SentryTimeToDisplayTracker.h"
+#import "SentryCurrentDate.h"
+#import "SentrySpan.h"
+#import "SentrySpanContext.h"
+#import "SentrySpanId.h"
+#import "SentrySpanOperations.h"
+#import "SentrySwift.h"
+
+#if SENTRY_HAS_UIKIT
+
+@interface
+SentryTimeToDisplayTracker ()
+
+@property (nonatomic, strong) NSDate *startDate;
+@property (nonatomic, strong) SentrySpan *initialDisplaySpan;
+@property (nonatomic, strong) NSDate *fullDisplayDate;
+@property (nonatomic, strong) NSString *controllerName;
+@property (nonatomic) BOOL waitForFullDisplay;
+
+@end
+
+@implementation SentryTimeToDisplayTracker
+
+- (instancetype)initForController:(UIViewController *)controller
+               waitForFullDisplay:(BOOL)waitForFullDisplay
+{
+    if (self = [super init]) {
+        self.startDate = [SentryCurrentDate date];
+        self.controllerName = [SwiftDescriptor getObjectClassName:controller];
+        self.waitForFullDisplay = waitForFullDisplay;
+    }
+    return self;
+}
+
+- (void)installForTracer:(SentryTracer *)tracer
+{
+    self.initialDisplaySpan =
+        [tracer startChildWithOperation:SentrySpanOperationUILoadInitialDisplay
+                            description:[NSString stringWithFormat:@"%@ initial display",
+                                                  self.controllerName]];
+}
+
+- (void)reportInitialDisplay
+{
+    if (self.initialDisplaySpan.timestamp != nil) {
+        return;
+    }
+
+    self.initialDisplaySpan.timestamp = [SentryCurrentDate date];
+
+    if (self.fullDisplayDate != nil &&
+        [self.fullDisplayDate compare:self.initialDisplaySpan.timestamp] == NSOrderedAscending) {
+        self.fullDisplayDate = self.initialDisplaySpan.timestamp;
+    }
+
+    if (!self.waitForFullDisplay || self.fullDisplayDate != nil) {
+        // If this class is waiting for a full display, we don't finish the TTID span
+        // because this will make the tracer wait for its children which gives the user more time to
+        // report full display, since SentryTracer have a dead line timeout, eventually it will be
+        // finished.
```

Comment:
I get it, but if the user enables the options he wants to track the full screens.
Having a span with deadline_exceeded is a way to remind him he didn't call the API.
We can make any other consideration based on user's feedback

Reply:
I agree with @brustolin on this one. We should print a warning log and maybe set some flag on the session so that we can indicate it on their dashboard in some other way.

Reply:
> but if the user enables the options he wants to track the full screens

I have a problem it this assertion. 
Lets say a big project with dozens of screens, but the user only want TTFD for main big screens, now they need to go at every view controller and report full display. 
Otherwise every other transaction will have trash span.

Reply:
> they need to go at every view controller and report full display
Isn't this how it's going to work anyways, since they must call `SentrySDK.reportFullyDiplayed` from each of those?

Reply:
This is what we are expecting them to do. Particularly I find this annoying. But we agreed to keep the implementation similar to Android.